ui/lib/core/addon/components/form-field-groups.hbs

```diff
@@ -17,7 +17,7 @@
             />
           {{else}}
             {{! OTHERWISE WE'RE EDITING }}
-            {{#if (or (eq attr.name "name") (attr.options.editDisabled))}}
+            {{#if (or (eq attr.name "name") (eq attr.name "username") (attr.options.editDisabled))}}
```

Comment:
Adding the conditional here will make this input readonly for any other models with a `username` attribute that use this component. 

Instead I'd approach this using CSS. Inspecting the DOM we can see that `readonly` is applied to the input, but for some reason the styling is not. I'd find where this `.scss` class is defined in the codebase and apply the styling fix there. 
<img width="978" alt="Screenshot 2023-07-12 at 10 15 13 AM" src="https://github.com/hashicorp/vault/assets/68122737/ae37af8e-8327-49c8-aefb-3e5d9b7b2431">